### PR TITLE
feat(ui): added custom role map to project, component and release

### PR DESCRIPTION
**Pull request by maximilianhuber**
This PR adds the following features:
* displayed in details-view (as mailto-link) and edit view
* keys can be configured in sw360-portlet: sw360.properties
* edit-view: user can choose key out of drop down list and must enter valid email address per row
* Moderation request generation for role-property
* Diff display of properties of type map of set to set of strings
* merge for properties of type map of set to set of strings

### CODE REVIEW

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java [hunk 1]
```diff
@@ -145,8 +146,6 @@ else if (field == Release._Fields.MAINLINE_STATE)
             return getMainlineStatefromString(value);
         else if (field == Project._Fields.STATE)
             return getProjectStateFromString(value);
-        else if (field == Project._Fields.CLEARING_STATE)
```
Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java]: You are right. This looks wrong.

File: frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/tags/DisplayMapOfEmailStringSets.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2016.
```
Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/tags/DisplayMapOfEmailStringSets.java]: This code was created 2016.

**Comment by maximilianhuber** [comment]
@alex-evo @mcjaeger: the roles are now updated and the branch is rebased.

**Comment by mcjaeger** [comment]
not so important: actually sw360.properties with old values overwrites default values!

**Pull request merged**